pkg/yaml/yaml_test.go

```diff
@@ -0,0 +1,496 @@
+package yaml
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/kluctl/kluctl/v2/pkg/utils"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"testing/iotest"
+)
+
+type EmptyYamlConfig struct {
+}
+
+type SimpleYamlConfig struct {
+	Value string `yaml:"value"`
+}
+
+func TestReadYamlFile(t *testing.T) {
+	// Setup variables
+	existingEmptyYamlFileNamePattern := "test_read_yaml_file_existing_empty_*.yaml"
+	var existingEmptyYamlConf EmptyYamlConfig
+	nonExistingEmptyYamlFileName := "test_read_yaml_file_non_existing_empty.yaml"
+	var nonExistingEmptyYamlConf EmptyYamlConfig
+
+	// Create directory for test files
+	createTestDir()
+	defer cleanupTestDir()
+
+	// Setup temporary file
+	file, err := CreateTempFile(existingEmptyYamlFileNamePattern)
```

Comment:
No need to use CreateTempFile() when using t.TempDir(), as you can use a static file name then